File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let arr = priorities.map((data, index) => [data, index])
    let complate = []
    let answer = 0
    
    

    while(arr.length > 0) {
        // 현재 값
        let cur = arr.shift()
        
        // 1개 남으면 종료
        if(arr.length === 0) {
            complate.push(cur)
            break
        }
        
        // 남은 값 중 최대 값
        let max = Math.max(...arr.map((data) => data[0]))
        
        // 작업 순서대로 재정렬
        if(cur[0] < max) {
            arr.push(cur)
        } else if (cur[0] >= max) {
            complate.push(cur)
        } 
    }
    
     complate.forEach((data, index) => {
        if(data[1] === location) {
            answer = index + 1 
        }
    })
    
    return answer
    
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    
    dq = deque([val, idx] for idx, val in enumerate(priorities))
    
    count = 0
    
    while dq:
        first_task = max(dq, key = lambda x :x[0])[0]
        task = dq.popleft()

        if task[0] >= first_task:
            count += 1
            if task[1] == location:
                return count
        else:
            dq.append(task)
            
    return count
        
        
        
    
    